allennlp/commands/diff.py

```diff
@@ -287,9 +227,29 @@ def one(idx):
     assert False, "Could not find edit script"
 
 
+def _get_checkpoint_path(checkpoint: str, checkpoint_type: str) -> str:
+    if checkpoint_type == "file":
+        if checkpoint.endswith(".tar.gz"):
+            return cached_path(checkpoint + "!weights.th", extract_archive=True)
+        elif ".tar.gz!" in checkpoint:
+            return cached_path(checkpoint, extract_archive=True)
+        else:
+            return cached_path(checkpoint)
+    elif checkpoint_type == "huggingface":
+        from transformers.file_utils import (
+            hf_bucket_url,
+            WEIGHTS_NAME,
+            cached_path as hf_cached_path,
+        )
```

Comment:
Can this go away when https://github.com/allenai/allennlp/pull/5052 is merged? If so, I'd rather wait. #5052 looks like it's almost ready for merging.

Reply:
In fact, if that is so, the whole notion of a checkpoint type can go away, right?